tests/helpers/test_device_registry.py

```diff
@@ -490,6 +495,126 @@ async def test_migration_1_2_to_1_3(hass, hass_storage):
                     "model": None,
                     "name_by_user": None,
                     "name": None,
+                    "serial_number": None,
+                    "sw_version": None,
+                    "via_device_id": None,
+                },
+            ],
+            "deleted_devices": [],
+        },
+    }
+
+
+@pytest.mark.parametrize("load_registries", [False])
+async def test_migration_1_3_to_1_4(hass, hass_storage):
+    """Test migration from version 1.3 to 1.4."""
+    hass_storage[device_registry.STORAGE_KEY] = {
+        "version": 1,
+        "minor_version": 3,
+        "key": device_registry.STORAGE_KEY,
+        "data": {
+            "devices": [
+                {
+                    "area_id": None,
+                    "config_entries": ["1234"],
+                    "configuration_url": None,
+                    "connections": [["Zigbee", "01.23.45.67.89"]],
+                    "disabled_by": None,
+                    "entry_type": "service",
+                    "hw_version": "hw_version",
+                    "id": "abcdefghijklm",
+                    "identifiers": [["serial", "12:34:56:AB:CD:EF"]],
```

Comment:
`"serial"` here is just a coincidence, right? It's not a requirement of this PR, right?

Reply:
That's indeed just a concidence 👍

Reply:
This should have rang a bell IMHO, as a serial number IS an identifier.